# Big Data for Biologists: Decoding Genomic Function- Class 8

## How can you learn about the function of a gene product by studying other organisms?  

##  Learning Objectives
***Students should be able to***
 <ol>
 <li>   </li>
 <li>   </li>

In [2]:
# magic! (don't worry about this)
%matplotlib inline
%load_ext autoreload
%autoreload 2
import matplotlib.pyplot as plt

import sys
sys.path.append('../helpers')
from kmeans_helpers import * 
from helpers import * 

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In the previous tutorial, we performed k-means clustering to group gene expression samples by tissue. Now, we are interested in similar groups of genes. We can flip the axis used for clustering to find k-means cluster of related genes. 

In [3]:
from sklearn.cluster import KMeans
from sklearn.metrics import silhouette_score,silhouette_samples

In [6]:
import pandas as pd 
systems_subset=["Blood","Embryonic","Immune","Respiratory"]
data,batches=load_rnaseq_data(systems_subset=systems_subset)
data.shape

(1543, 116)

First, we will run all 1543 differential genes through GORilla to determine if there are any significantly enriched GO terms. Write the genes to an output file called 'all_genes.txt' and load this file into the GORilla window below. Select "All" for Ontology to search the Cellular Component, Biological Process, and Molecular Function ontologies. 


In [21]:
gene_names=data.index
gene_names[0:10]

Index([u'ENSG00000269416.1', u'ENSG00000008517.12', u'ENSG00000158966.9',
       u'ENSG00000167768.4', u'ENSG00000147257.9', u'ENSG00000169992.5',
       u'ENSG00000143248.8', u'ENSG00000232301.1', u'ENSG00000083857.9',
       u'ENSG00000165272.10'],
      dtype='object')

In [22]:
outf=open('all_genes.txt','w')
outf.write('\n'.join(gene_names))

In [23]:
!head all_genes.txt


ENSG00000269416.1
ENSG00000008517.12
ENSG00000158966.9
ENSG00000167768.4
ENSG00000147257.9
ENSG00000169992.5
ENSG00000143248.8
ENSG00000232301.1
ENSG00000083857.9
ENSG00000165272.10


In [24]:
from IPython.display import IFrame

IFrame('http://cbl-gorilla.cs.technion.ac.il/',height=500,width=1000)

It looks like GORilla is not able to recognize the ENSEMBL gene id's. We must convert them to the official gene symbol format. We can perform this conversion with the DAVID analysis tool, which will also identify GO terms and pathways. 

In [ ]:
https://david.ncifcrf.gov/